src/Handler/ResultAcknowledgementHandler.php

```diff
@@ -63,9 +63,24 @@ protected function acknowledge(
     ) {
         if (call_user_func($this->validator, $result) === true) {
             $this->handler->acknowledge($message, $adapter, $result);
+        } else {
+            $this->handler->reject($message, $adapter, $result);
```

Comment:
Is it a little strange that you call `acknowledge` and you message might get rejected? Given there are two possible outcomes, would it make sense to have some way of knowing which path was taken?

(I'm not familiar with the context, so this might be fine).

Reply:
The whole way this library works is a little whack. The result handler is currently the only way to say yay/nay for a message without throwing an exception.